keyboards/bastardkb/dilemma/4x6_4/4x6_4.c

```diff
@@ -57,3 +64,59 @@ bool encoder_update_kb(uint8_t index, bool clockwise) {
     return true;
 }
 #endif // ENCODER_ENABLE
+
+#ifdef RGB_MATRIX_ENABLE
+HSV _get_hsv_for_layer_index(uint8_t layer) {
+    switch (layer) {
+        case 1:
+            return (HSV){HSV_BLUE};
+        case 2:
+            return (HSV){HSV_AZURE};
+        case 3:
+            return (HSV){HSV_ORANGE};
+        case 4:
+            return (HSV){HSV_GREEN};
+        case 5:
+            return (HSV){HSV_TEAL};
+        case 6:
+            return (HSV){HSV_PURPLE};
+        default:
+            return (HSV){HSV_RED};
+    };
+}
+
+// Layer state indicator
+bool rgb_matrix_indicators_advanced_kb(uint8_t led_min, uint8_t led_max) {
+    if (!rgb_matrix_indicators_advanced_user(led_min, led_max)) {
+        return false;
+    }
+    // Set indicator LEDs to red if caps lock is enabled
+    if (host_keyboard_led_state().caps_lock) {
+        for (int i = led_min; i <= led_max; i++) {
+            if (HAS_FLAGS(g_led_config.flags[i], LED_FLAG_MODIFIER)) {
+                // set modifier-flagged LEDs to a pure a configured interval brighter than the current brightness, clamped to 255 (ie. uint8_t max value).
+                rgb_matrix_set_color(i, MIN(rgb_matrix_get_val() + CAPSLOCK_INDICATOR_BRIGHTNESS_INC, 255), 0, 0);
+            }
+        }
+    }
+
+    // Set underglow to a solid color for highest active layer apart from the base layer.
+    const uint8_t layer = get_highest_layer(layer_state);
+    if (layer > 0) {
+        HSV hsv = _get_hsv_for_layer_index(layer);
+
+        if (hsv.v > rgb_matrix_get_val()) {
```

Comment:
I'm not sure if I misread this condition before or if it has changed, but this makes even less sense to me now.

Are we not raising the brightness if the value is lower than the user configured one? This means that we only raise the brightness if its already passed the default one.

Is this expected, or should we remove the condition altogether?

Reply:
It only ever made a difference on layer 5, since only teal (and torquoise, which we don't use) doesn't come with maximum brightness. You're right that we should boost it all the same; I've removed the condition and verified in hardware that it looks fitting this way even on layer 5.